### PR TITLE
fix(regex 942240) - disassembled data file

### LINKED ISSUE #2859: Dissasemble regexes in rules 942230 and 942240
### Motivation

All regexes should be generated using regexp-assembly.

### Proposed solution

Dissasemble the regexes in rule 942230 and 942240.

### PR DESCRIPTION
**Pull request by franbuehler**
This PR closes #2859 by disassembling data file of 942240. @theseion confirmed that it doesn't make sense to disassemble 94230. So I did not touch this one.

The generated rule looks a bit different (also according to `regexp-assemble.py compare`) but it's still valid.

### CODE REVIEW

File: util/regexp-assemble/data/942240.data [hunk 1]
```diff
@@ -3,8 +3,30 @@
...
-[\"'`];*?\s*?waitfor\s+delay\s+[\"'`]
-[\"'`];.*?:\s*?goto
+##! Helpers
+##!> template alter alter\s*?\w+.*?
+
+##!> template set \s+set\s+\w+
+
+##!> template ticks [\"'`]
+
+##!> template waitfor *?\s*?waitfor\s+
```
Comment (theseion) [inline util/regexp-assemble/data/942240.data:13]: Looking at this expression, I have the feeling that the `.` got lost at some point. I don't see the reason to match `;*`, which would be included in `.*` anyway. By making this change, you can also move the `;` up to after `{{ticks}}`, below.

```suggestion
##!> template waitfor .*?\s*?waitfor\s+
```

File: util/regexp-assemble/data/942240.data [hunk 2]
```diff
@@ -3,8 +3,30 @@
...
+{{alter}}
+  ##!=>
+char
+character
+  ##!=>
+{{set}}
+  ##!<
+
+  ##!> assemble
+{{ticks}}
```
Comment (theseion) [inline util/regexp-assemble/data/942240.data:26]:
```suggestion
{{ticks}};
```

File: util/regexp-assemble/data/942240.data [hunk 3]
```diff
@@ -3,8 +3,30 @@
...
+  ##!=>
+;{{waitfor}}time\s+[\"'`]
```
Comment (theseion) [inline util/regexp-assemble/data/942240.data:28]:
```suggestion
{{waitfor}}time\s+[\"'`]
```

File: util/regexp-assemble/data/942240.data [hunk 4]
```diff
@@ -3,8 +3,30 @@
...
+;{{waitfor}}delay\s+[\"'`]
```
Comment (theseion) [inline util/regexp-assemble/data/942240.data:29]:
```suggestion
{{waitfor}}delay\s+[\"'`]
```

File: util/regexp-assemble/data/942240.data [hunk 5]
```diff
@@ -3,8 +3,30 @@
...
+;.*?:\s*?goto
```
Comment (theseion) [inline util/regexp-assemble/data/942240.data:30]:
```suggestion
.*?:\s*?goto
```

**Review by theseion — CHANGES_REQUESTED** [review]

File: util/regexp-assemble/data/942240.data (hunk 1 above)
Comment (franbuehler) [reply util/regexp-assemble/data/942240.data]: No, the problem was that I split the following regex into two parts: `;*?` a lazy match of `;`.
I tried to fix it again and hopefully the tests will succeed this time.

**Review by theseion — APPROVED** [review]
Thanks!

**Pull request merged**